### PR TITLE
Upgrade API version in test suite to 2017-04-06

### PR DESCRIPTION
**Pull request by brandur-stripe**
Upgrades the API version to the point where the transfers/payouts split gets included.

### CODE REVIEW

**Comment by brandur-stripe** [comment]
5 failures here relating to use `recipient` in transfers:

```
There were 5 errors:
1) Stripe\TransferReversalTest::testList
Stripe\Error\InvalidRequest: Received unknown parameters: description, recipient
/home/travis/build/stripe/stripe-php/lib/ApiRequestor.php:110
/home/travis/build/stripe/stripe-php/lib/ApiRequestor.php:275
/home/travis/build/stripe/stripe-php/lib/ApiRequestor.php:65
/home/travis/build/stripe/stripe-php/lib/ApiResource.php:120
/home/travis/build/stripe/stripe-php/lib/ApiResource.php:159
/home/travis/build/stripe/stripe-php/lib/Transfer.php:69
/home/travis/build/stripe/stripe-php/tests/TestCase.php:93
/home/travis/build/stripe/stripe-php/tests/TransferReversalTest.php:10
2) Stripe\TransferTest::testCreate
Stripe\Error\InvalidRequest: Received unknown parameter: recipient
/home/travis/build/stripe/stripe-php/lib/ApiRequestor.php:110
/home/travis/build/stripe/stripe-php/lib/ApiRequestor.php:275
/home/travis/build/stripe/stripe-php/lib/ApiRequestor.php:65
/home/travis/build/stripe/stripe-php/lib/ApiResource.php:120
/home/travis/build/stripe/stripe-php/lib/ApiResource.php:159
/home/travis/build/stripe/stripe-php/lib/Transfer.php:69
/home/travis/build/stripe/stripe-php/tests/TransferTest.php:16
3) Stripe\TransferTest::testRetrieve
Stripe\Error\InvalidRequest: Received unknown parameter: recipient
/home/travis/build/stripe/stripe-php/lib/ApiRequestor.php:110
/home/travis/build/stripe/stripe-php/lib/ApiRequestor.php:275
/home/travis/build/stripe/stripe-php/lib/ApiRequestor.php:65
/home/travis/build/stripe/stripe-php/lib/ApiResource.php:120
/home/travis/build/stripe/stripe-php/lib/ApiResource.php:159
/home/travis/build/stripe/stripe-php/lib/Transfer.php:69
/home/travis/build/stripe/stripe-php/tests/TransferTest.php:29
4) Stripe\TransferTest::testTransferUpdateMetadata
Stripe\Error\InvalidRequest: Received unknown parameter: recipient
/home/travis/build/stripe/stripe-php/lib/ApiRequestor.php:110
/home/travis/build/stripe/stripe-php/lib/ApiRequestor.php:275
/home/travis/build/stripe/stripe-php/lib/ApiRequestor.php:65
/home/travis/build/stripe/stripe-php/lib/ApiResource.php:120
/home/travis/build/stripe/stripe-php/lib/ApiResource.php:159
/home/travis/build/stripe/stripe-php/lib/Transfer.php:69
/home/travis/build/stripe/stripe-php/tests/TransferTest.php:62
5) Stripe\TransferTest::testTransferUpdateMetadataAll
Stripe\Error\InvalidRequest: Received unknown parameter: recipient
/home/travis/build/stripe/stripe-php/lib/ApiRequestor.php:110
/home/travis/build/stripe/stripe-php/lib/ApiRequestor.php:275
/home/travis/build/stripe/stripe-php/lib/ApiRequestor.php:65
/home/travis/build/stripe/stripe-php/lib/ApiResource.php:120
/home/travis/build/stripe/stripe-php/lib/ApiResource.php:159
/home/travis/build/stripe/stripe-php/lib/Transfer.php:69
/home/travis/build/stripe/stripe-php/tests/TransferTest.php:80
FAILURES!
Tests: 154, Assertions: 347, Errors: 5.
```

File: tests/TestCase.php [hunk 1]
```diff
@@ -77,7 +77,7 @@ protected static function createTestCharge(array $attributes = array())
     /**
      * Create a valid test charge.
```
Comment (remi-stripe) [inline tests/TestCase.php]: a bit unrelated but can we fix that comment while we're at it?

**Comment by brandur-stripe** [comment]
Released as 4.6.0.

**Pull request merged**